pages/upload.tsx

```diff
@@ -12,15 +13,44 @@ function UploadScreenWrapper({ onUpload }: { onUpload: () => void }) {
   const { recording } = useGetRecording(recordingId);
   // Make sure to get the user's settings and workspaces before showing the upload screen.
   const { userSettings, loading: userSettingsLoading } = useGetUserSettings();
-  const { workspaces, loading: pendingWorkspacesLoading } = useGetNonPendingWorkspaces();
+  const { workspaces, loading: workspacesLoading } = useGetNonPendingWorkspaces();
+  const initializeRecording = useInitializeRecording();
+  const [initializing, setInitializing] = useState(false);
 
   useEffect(() => {
     if (recording?.isInitialized) {
       window.onbeforeunload = null;
     }
   });
 
-  if (userSettingsLoading || pendingWorkspacesLoading) {
+  useEffect(() => {
+    async function initialize() {
+      await initializeRecording({
+        variables: { recordingId, title: recording?.title || "Untitled", workspaceId: null },
+      });
+      onUpload();
+    }
+    if (
+      recording?.isInitialized === false &&
+      !initializing &&
+      !workspacesLoading &&
+      workspaces.length === 0 &&
+      getRecordingTarget(recording.buildId ?? "unknown") !== RecordingTarget.gecko
```

Comment:
Wish we had a better, capabilities based way of detecting this. (Although maybe we want to also bypass this screen for Chromium recordings that are streamed in the background while being recorded?)